File: src/main/kotlin/cn/har01d/notebook/core/SiteConfig.kt
package cn.har01d.notebook.core

import cn.har01d.notebook.core.config.QiniuProperties

data class SiteConfig(
        val siteName: String,
        val brandColor: String,
        val qrCode: String,
        val icpBeian: String,
        val govBeian: String,
        val showViews: Boolean,
        val showWords: Boolean,
        val enableAudit: Boolean,
        val enableComment: Boolean,
        val enableFileUpload: Boolean,
        val enableImageUpload: Boolean,
        val enableSignup: Boolean,
        val enableHeartbeat: Boolean,
        val qiniu: QiniuProperties,
)

File: web-ui/src/models/SiteConfig.ts
export class QiniuProperties {
  enabled: boolean = false
  accessKey: string = ''
  secretKey: string = ''
  bucket: string = ''
  domain: string = ''
}

export class SiteConfig {
  siteName: string = 'Notebook'
  brandColor: string = 'teal'
  qrCode: string = ''
  icpBeian: string = ''
  govBeian: string = ''
  showViews: boolean = false
  showWords: boolean = false
  enableAudit: boolean = true
  enableComment: boolean = true
  enableFileUpload: boolean = true
  enableImageUpload: boolean = true
  enableSignup: boolean = true
  enableHeartbeat: boolean = true
  enableEncrypt: boolean = false
  secretKey: string = ''
  qiniu: QiniuProperties = new QiniuProperties()
}

export class DveConfig {
  developer: string = 'Har01d'
  version: string = '0.1.0'
}
